### DISCUSSION TITLE
Hardhat Version

### DISCUSSION THREAD

**Discussion by Jsoooooooo** [Jeiwan/uniswapv3-book#12] [General]
thanks for the fantastic version of this book. I am not sure wether it is possible to provide with a hardhat version of deploy and test or not ?

thanks,

**Comment by Jeiwan** [comment]
The original code was tested and deployed using Hardhat:
https://github.com/Uniswap/v3-core

For the book, I chose Foundry deliberately since it's faster, easier to setup and use, and it allows to write tests in Solidity.

**Comment by Jsoooooooo** [comment]
Again, thanks a lot for the fantastic book.

Kind regards,

**Comment by Jsoooooooo** [comment]
Dear Ivan,


I am an amateur web3 developer. Since there are many parameters while deploying the v3 Book contracts, I am a bit confused while deploying by hardhat. Could you please provide a hardhat version of V3 book deployment?


Many thanks,
David

**Comment by Jsoooooooo** [comment]
Hello,

Sorry to bother you again. I stuck with deploying the Pool contract with hardhat for couple of days, and met an error  term  “ function call to a non-contract account”. 

I think the solidity code origin that occurs the error is listed as below, while I didn’t find any answers on how to pass parameters while deploying.



Here is my deploy script:


Could you please teach me on how to solve the problem?

Many thanks,

**Reply by Jeiwan** [reply]
The script doesn't show up. Save it to a gist and paste the link here.

“function call to a non-contract account” most likely means that you're trying to call a function on an address that doesn't have a contract deployed at. Probably, you're trying to call a contract on a wrong address.

**Comment by Jsoooooooo** [comment]
Thanks a lot.

Many thanks.
